src/main/java/net/sf/jabref/model/database/DuplicationChecker.java

```diff
@@ -2,62 +2,43 @@
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Optional;
 
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
+import net.sf.jabref.model.database.event.EntryAddedEvent;
+import net.sf.jabref.model.database.event.EntryRemovedEvent;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.event.FieldChangedEvent;
+
+import com.google.common.eventbus.Subscribe;
 
 /**
  * Determines which bibtex cite keys are duplicates in a single {@link BibDatabase}.
  */
-class DuplicationChecker {
-
-    private static final Log LOGGER = LogFactory.getLog(DuplicationChecker.class);
+public class DuplicationChecker {
 
-    // use a map instead of a set since i need to know how many of each key is in there
+    /** use a map instead of a set since I need to know how many of each key is in there */
     private final Map<String, Integer> allKeys = new HashMap<>();
 
+
     /**
-     * Usage:
-     * <br>
-     * isDuplicate=checkForDuplicateKeyAndAdd( null, b.getKey() , issueDuplicateWarning);
-     *
-     * If the newkey already exists and is not the same as oldkey it will give a warning
-     * else it will add the newkey to the to set and remove the oldkey
-     *
-     * @return true, if there is a duplicate key, else false
+     * Checks if there is more than one occurrence of this key
      */
-    public boolean checkForDuplicateKeyAndAdd(String oldKey, String newKey) {
+    public boolean isDuplicateExisting(String citeKey) {
+        return getNumberOfKeyOccurrences(citeKey) > 1;
+    }
 
-        boolean duplicate = false;
-        if (oldKey == null) { // No old key
-            duplicate = addKeyToSet(newKey);
-        } else {
-            if (oldKey.equals(newKey)) {// were OK because the user did not change keys
-                duplicate = false;
-            } else {
-                removeKeyFromSet(oldKey); // Get rid of old key
-                if (newKey != null) { // Add new key if any
-                    duplicate = addKeyToSet(newKey);
-                }
-            }
-        }
-        if (duplicate) {
-            LOGGER.warn("Warning there is a duplicate key: " + newKey);
-        }
-        return duplicate;
+    /**
+     * Checks if there is more than one occurrence of this key
+     */
+    public boolean isDuplicateExisting(Optional<String> citeKey) {
```

Comment:
Please don't use optionals as params. They should only be used as return values. See https://github.com/JabRef/jabref/pull/1577#r75998843